trieste/models/model_interfaces.py

```diff
@@ -274,13 +274,18 @@ def update(self, dataset: Dataset):
         assert dataset.observations.shape[-1] == y.shape[-1]
         data = (dataset.query_points, dataset.observations)
         num_data = data[0].shape[0]
-        num_latent_gps = model.num_latent_gps
+
+        new_q_mu, new_q_var = self.model.predict_f(dataset.query_points, full_cov=True)
+
+        new_q_sqrt = tf.linalg.cholesky(
+            new_q_var
+            + gpflow.config.default_jitter() * (tf.eye(num_data, dtype=new_q_var.dtype)[None])
```

Comment:
FYI: gpflow has `gpflow.utilities.ops.eye(num, value=gpflow.config.default_jitter())` that you could use for this.